From patchwork Fri Feb 12 01:06:03 2021
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Sean Christopherson <seanjc@google.com>
X-Patchwork-Id: 12084563
Return-Path: <kvm-owner@kernel.org>
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on
	aws-us-west-2-korg-lkml-1.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-26.3 required=3.0 tests=BAYES_00,DKIMWL_WL_MED,
	DKIM_SIGNED,DKIM_VALID,DKIM_VALID_AU,HEADER_FROM_DIFFERENT_DOMAINS,
	INCLUDES_CR_TRAILER,INCLUDES_PATCH,MAILING_LIST_MULTI,SPF_HELO_NONE,SPF_PASS,
	USER_AGENT_GIT,USER_IN_DEF_DKIM_WL autolearn=ham autolearn_force=no
	version=3.4.0
Received: from mail.kernel.org (mail.kernel.org [198.145.29.99])
	by smtp.lore.kernel.org (Postfix) with ESMTP id C73C3C433E6
	for <kvm@archiver.kernel.org>; Fri, 12 Feb 2021 01:09:04 +0000 (UTC)
Received: from vger.kernel.org (vger.kernel.org [23.128.96.18])
	by mail.kernel.org (Postfix) with ESMTP id 8FEAA64E5F
	for <kvm@archiver.kernel.org>; Fri, 12 Feb 2021 01:09:04 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S230365AbhBLBIn (ORCPT <rfc822;kvm@archiver.kernel.org>);
        Thu, 11 Feb 2021 20:08:43 -0500
Received: from lindbergh.monkeyblade.net ([23.128.96.19]:35720 "EHLO
        lindbergh.monkeyblade.net" rhost-flags-OK-OK-OK-OK) by vger.kernel.org
        with ESMTP id S230180AbhBLBHJ (ORCPT <rfc822;kvm@vger.kernel.org>);
        Thu, 11 Feb 2021 20:07:09 -0500
Received: from mail-yb1-xb4a.google.com (mail-yb1-xb4a.google.com
 [IPv6:2607:f8b0:4864:20::b4a])
        by lindbergh.monkeyblade.net (Postfix) with ESMTPS id 1DCCDC0617A7
        for <kvm@vger.kernel.org>; Thu, 11 Feb 2021 17:06:13 -0800 (PST)
Received: by mail-yb1-xb4a.google.com with SMTP id g17so5365814ybh.4
        for <kvm@vger.kernel.org>; Thu, 11 Feb 2021 17:06:13 -0800 (PST)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=google.com; s=20161025;
        h=sender:reply-to:date:in-reply-to:message-id:mime-version:references
         :subject:from:to:cc;
        bh=Fhv1Q9NSxxwZmv6ZVWXSl3ZrCdVknKiV94qk2PWj5KY=;
        b=M6sNjp71JNXOPH26XcRa1YdrRCuZtL4Py6H/Hy2tsrpRa6Zhg12Hr8v7orLO2dXdjh
         A+KWhDigFcqg5KAxt2kGBBOqPgYElc0VsDwlZA2Pu6bkeJqiNP+cUu3ccFBa3Ije+s7t
         /jBtoGykJDuAQW+gcKxrwtnONoU/iok5WjANsDmK0mVOR3QwWPLdOI/9/6lIYrWX/kbj
         qKc3QHkhd8jippcsGugpP7eh6otMsYLmveRp715jRaN2Q9Er58/hQZKh9Hb69nKpBwpG
         bRWNJfX7J7x6SC41n4v/QE6e9r+toCi11KhZ7zao+fndrhyuGgQoIlfL3htU2i46BJjW
         EZ9g==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20161025;
        h=x-gm-message-state:sender:reply-to:date:in-reply-to:message-id
         :mime-version:references:subject:from:to:cc;
        bh=Fhv1Q9NSxxwZmv6ZVWXSl3ZrCdVknKiV94qk2PWj5KY=;
        b=TB5aMpmkTVfybXWlrLouwQtCTZEvr3oecc+PdyVbA0siXX7cDAB2cQ3CGDy3qM6PXd
         AqUsqbWCxkA8xakuvVaBiCDeMS9yB2v3B1at6sevht0OJOzxELaRcJbKsCWUb/7VBqjN
         DHRPdVG7di7VUQUu21Q1hTdxVnxLdIva6UuFh9HVtqdqAtFGrlNiN4xW/3pejxsTY0oL
         5MVHbdGUQaL7JZ1RQwhk1S+Jft2OWGyDG3XMdP8w0jI2cLvJ3sRkN0vvgu6EnQwuPqdJ
         jz+FivtYRjF+BOauYP82xdyEmSEgd1YajyTn9ikpfQ+X8Un5u8Ehnvtekcb6V9n4JtRx
         qEkw==
X-Gm-Message-State: AOAM530QjVdTHehVlCVpoxgNF8PgwOwtKRbvlwe6qp2aHoGswEj0p+1s
        V3c/GxF6CpGbHBpb8MPigWs/OTdXTU0=
X-Google-Smtp-Source: 
 ABdhPJxYLiR7Gj05G/CPO+CWigXBb8GTWY0DxsO//rw0ZXfWZiF/u6c32DhhVqZtgdkTylmeNAnIJoMIuj0=
Sender: "seanjc via sendgmr" <seanjc@seanjc798194.pdx.corp.google.com>
X-Received: from seanjc798194.pdx.corp.google.com
 ([2620:15c:f:10:f588:a708:f347:3ebb])
 (user=seanjc job=sendgmr) by 2002:a25:908f:: with SMTP id
 t15mr782326ybl.47.1613091972358;
 Thu, 11 Feb 2021 17:06:12 -0800 (PST)
Reply-To: Sean Christopherson <seanjc@google.com>
Date: Thu, 11 Feb 2021 17:06:03 -0800
In-Reply-To: <20210212010606.1118184-1-seanjc@google.com>
Message-Id: <20210212010606.1118184-2-seanjc@google.com>
Mime-Version: 1.0
References: <20210212010606.1118184-1-seanjc@google.com>
X-Mailer: git-send-email 2.30.0.478.g8a0d178c01-goog
Subject: [kvm-unit-tests PATCH 1/4] x86: Remove PCID test that INVPCID isn't
 enabled without PCID
From: Sean Christopherson <seanjc@google.com>
To: Paolo Bonzini <pbonzini@redhat.com>
Cc: kvm@vger.kernel.org, Sean Christopherson <seanjc@google.com>
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org

Drop the test that verifies INVPCID isn't enabled in the guest without
PCID also being enabled.  SVM allows !PCID && INVPCID, and VMX will soon
follow.

https://lkml.kernel.org/r/20210212003411.1102677-1-seanjc@google.com

Signed-off-by: Sean Christopherson <seanjc@google.com>
---
 x86/pcid.c | 8 --------
 1 file changed, 8 deletions(-)

diff --git a/x86/pcid.c b/x86/pcid.c
index a8dc8cb..ee0b726 100644
--- a/x86/pcid.c
+++ b/x86/pcid.c
@@ -26,12 +26,6 @@ static int invpcid_checking(unsigned long type, void *desc)
     return exception_vector();
 }
 
-static void test_cpuid_consistency(int pcid_enabled, int invpcid_enabled)
-{
-    int passed = !(!pcid_enabled && invpcid_enabled);
-    report(passed, "CPUID consistency");
-}
-
 static void test_pcid_enabled(void)
 {
     int passed = 0;
@@ -135,8 +129,6 @@ int main(int ac, char **av)
     if (this_cpu_has(X86_FEATURE_INVPCID))
         invpcid_enabled = 1;
 
-    test_cpuid_consistency(pcid_enabled, invpcid_enabled);
-
     if (pcid_enabled)
         test_pcid_enabled();
     else

From patchwork Fri Feb 12 01:06:04 2021
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Sean Christopherson <seanjc@google.com>
X-Patchwork-Id: 12084561
Return-Path: <kvm-owner@kernel.org>
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on
	aws-us-west-2-korg-lkml-1.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-26.3 required=3.0 tests=BAYES_00,DKIMWL_WL_MED,
	DKIM_SIGNED,DKIM_VALID,DKIM_VALID_AU,HEADER_FROM_DIFFERENT_DOMAINS,
	INCLUDES_CR_TRAILER,INCLUDES_PATCH,MAILING_LIST_MULTI,SPF_HELO_NONE,SPF_PASS,
	USER_AGENT_GIT,USER_IN_DEF_DKIM_WL autolearn=ham autolearn_force=no
	version=3.4.0
Received: from mail.kernel.org (mail.kernel.org [198.145.29.99])
	by smtp.lore.kernel.org (Postfix) with ESMTP id E3E2AC433E9
	for <kvm@archiver.kernel.org>; Fri, 12 Feb 2021 01:09:04 +0000 (UTC)
Received: from vger.kernel.org (vger.kernel.org [23.128.96.18])
	by mail.kernel.org (Postfix) with ESMTP id AC68D64E39
	for <kvm@archiver.kernel.org>; Fri, 12 Feb 2021 01:09:04 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S230369AbhBLBIz (ORCPT <rfc822;kvm@archiver.kernel.org>);
        Thu, 11 Feb 2021 20:08:55 -0500
Received: from lindbergh.monkeyblade.net ([23.128.96.19]:35732 "EHLO
        lindbergh.monkeyblade.net" rhost-flags-OK-OK-OK-OK) by vger.kernel.org
        with ESMTP id S230185AbhBLBHM (ORCPT <rfc822;kvm@vger.kernel.org>);
        Thu, 11 Feb 2021 20:07:12 -0500
Received: from mail-qk1-x74a.google.com (mail-qk1-x74a.google.com
 [IPv6:2607:f8b0:4864:20::74a])
        by lindbergh.monkeyblade.net (Postfix) with ESMTPS id 73505C0617AB
        for <kvm@vger.kernel.org>; Thu, 11 Feb 2021 17:06:15 -0800 (PST)
Received: by mail-qk1-x74a.google.com with SMTP id o16so6059011qkj.15
        for <kvm@vger.kernel.org>; Thu, 11 Feb 2021 17:06:15 -0800 (PST)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=google.com; s=20161025;
        h=sender:reply-to:date:in-reply-to:message-id:mime-version:references
         :subject:from:to:cc;
        bh=cEJRTvR1R8mp+nZlLc5IDPe77GXhuZ/MSPi9P8LN9jY=;
        b=abpETcZ2hb0WlkAri6vgnfiqY8Xku/wFJFhluLYbZ5edXXT/xQWniUgmbpkcw5Be65
         mCx67We2tMxTzLSJFw5T89OinhT6RO78GC9hEz6ox1szAjwhJoj6rnMs9Y485+Ap81L3
         zfN6pQKB/Aq9RGIip0JmRwBEK7SbLPTcj+iCO0EaWfGP83z3gjaZ64l/36fTpDbOfOWR
         rHE61zRhTB2vaT/cigBMjacAIoSv3ZRy57Ic5OZrydBtV+ehWYEjXeknwfczZtrzIgwd
         JgywwHG/kWqn0LR7+wTyNEedMbg5Ntp69ply0kGkUOHgUGV8cKm2aaMYd7tsrBHWd1dB
         3/uQ==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20161025;
        h=x-gm-message-state:sender:reply-to:date:in-reply-to:message-id
         :mime-version:references:subject:from:to:cc;
        bh=cEJRTvR1R8mp+nZlLc5IDPe77GXhuZ/MSPi9P8LN9jY=;
        b=DnFdt99l0FuIe4DP3TnIiHNfL2hcBHe+ium0Xmw2GBPAk+1jQR5LkgcNdc2qaDr45H
         msZE+bhzPreswjCX+zDtQJqVgHB//j6FILcLyyPD//zQspspqE0qBrULW3cQbinTVu6H
         hQnKLd7Z/2ApQBUb4yL+W+vDhCmTjE/b3MvNhzGjr/3G8aTFh9X1WV/RGU2yq9Np3dNs
         7wrQLrq3n0Ujmi5VkYtbFU3hkui/HVUi8aBPixpeEUsev/MbKQqHWY1/5BAJqgmKG9W0
         z6IJigSoBpRhQ9DSj/+8KbQgw0+Enyg5nB+is3owkBVuvvB4JZm917dUfpHTHQW4KgOg
         VTBw==
X-Gm-Message-State: AOAM530p1Y+dO2JOUQJkphCZemYqusAjABhuppIJsDlHn6YufoxpkiN9
        RfjrW2+UAyp3qbiyWTHtlgn5g3BmNOA=
X-Google-Smtp-Source: 
 ABdhPJz950wjkbLvEqaZkr7CuN6+xpFhkxiDLKvuDnPUCBT0ebF3wWGIz/wZ4KGMAdUw2ufbF++QSOXAT5Q=
Sender: "seanjc via sendgmr" <seanjc@seanjc798194.pdx.corp.google.com>
X-Received: from seanjc798194.pdx.corp.google.com
 ([2620:15c:f:10:f588:a708:f347:3ebb])
 (user=seanjc job=sendgmr) by 2002:a0c:b611:: with SMTP id
 f17mr546095qve.42.1613091974598;
 Thu, 11 Feb 2021 17:06:14 -0800 (PST)
Reply-To: Sean Christopherson <seanjc@google.com>
Date: Thu, 11 Feb 2021 17:06:04 -0800
In-Reply-To: <20210212010606.1118184-1-seanjc@google.com>
Message-Id: <20210212010606.1118184-3-seanjc@google.com>
Mime-Version: 1.0
References: <20210212010606.1118184-1-seanjc@google.com>
X-Mailer: git-send-email 2.30.0.478.g8a0d178c01-goog
Subject: [kvm-unit-tests PATCH 2/4] x86: Iterate over all INVPCID types in the
 enabled test
From: Sean Christopherson <seanjc@google.com>
To: Paolo Bonzini <pbonzini@redhat.com>
Cc: kvm@vger.kernel.org, Sean Christopherson <seanjc@google.com>
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org

Verify that INVPCID works with all types, even those that take a PCID,
when CR4.PCIDE=0.  So long as the target PCID=0, INVPCID is legal even
if CR4.PCIDE=0.

Signed-off-by: Sean Christopherson <seanjc@google.com>
---
 x86/pcid.c | 18 +++++++++++-------
 1 file changed, 11 insertions(+), 7 deletions(-)

diff --git a/x86/pcid.c b/x86/pcid.c
index ee0b726..64efd05 100644
--- a/x86/pcid.c
+++ b/x86/pcid.c
@@ -70,24 +70,28 @@ report:
 
 static void test_invpcid_enabled(void)
 {
-    int passed = 0;
+    int passed = 0, i;
     ulong cr4 = read_cr4();
     struct invpcid_desc desc;
     desc.rsv = 0;
 
-    /* try executing invpcid when CR4.PCIDE=0, desc.pcid=0 and type=1
+    /* try executing invpcid when CR4.PCIDE=0, desc.pcid=0 and type=0..3
      * no exception expected
      */
     desc.pcid = 0;
-    if (invpcid_checking(1, &desc) != 0)
-        goto report;
+    for (i = 0; i < 4; i++) {
+        if (invpcid_checking(i, &desc) != 0)
+            goto report;
+    }
 
-    /* try executing invpcid when CR4.PCIDE=0, desc.pcid=1 and type=1
+    /* try executing invpcid when CR4.PCIDE=0, desc.pcid=1 and type=0..1
      * #GP expected
      */
     desc.pcid = 1;
-    if (invpcid_checking(1, &desc) != GP_VECTOR)
-        goto report;
+    for (i = 0; i < 2; i++) {
+        if (invpcid_checking(i, &desc) != GP_VECTOR)
+            goto report;
+    }
 
     if (write_cr4_checking(cr4 | X86_CR4_PCIDE) != 0)
         goto report;

From patchwork Fri Feb 12 01:06:05 2021
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Sean Christopherson <seanjc@google.com>
X-Patchwork-Id: 12084565
Return-Path: <kvm-owner@kernel.org>
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on
	aws-us-west-2-korg-lkml-1.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-26.3 required=3.0 tests=BAYES_00,DKIMWL_WL_MED,
	DKIM_SIGNED,DKIM_VALID,DKIM_VALID_AU,HEADER_FROM_DIFFERENT_DOMAINS,
	INCLUDES_CR_TRAILER,INCLUDES_PATCH,MAILING_LIST_MULTI,SPF_HELO_NONE,SPF_PASS,
	USER_AGENT_GIT,USER_IN_DEF_DKIM_WL autolearn=ham autolearn_force=no
	version=3.4.0
Received: from mail.kernel.org (mail.kernel.org [198.145.29.99])
	by smtp.lore.kernel.org (Postfix) with ESMTP id E9D64C433DB
	for <kvm@archiver.kernel.org>; Fri, 12 Feb 2021 01:09:46 +0000 (UTC)
Received: from vger.kernel.org (vger.kernel.org [23.128.96.18])
	by mail.kernel.org (Postfix) with ESMTP id 8D34164E39
	for <kvm@archiver.kernel.org>; Fri, 12 Feb 2021 01:09:46 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S230394AbhBLBJK (ORCPT <rfc822;kvm@archiver.kernel.org>);
        Thu, 11 Feb 2021 20:09:10 -0500
Received: from lindbergh.monkeyblade.net ([23.128.96.19]:35738 "EHLO
        lindbergh.monkeyblade.net" rhost-flags-OK-OK-OK-OK) by vger.kernel.org
        with ESMTP id S230194AbhBLBHN (ORCPT <rfc822;kvm@vger.kernel.org>);
        Thu, 11 Feb 2021 20:07:13 -0500
Received: from mail-qv1-xf49.google.com (mail-qv1-xf49.google.com
 [IPv6:2607:f8b0:4864:20::f49])
        by lindbergh.monkeyblade.net (Postfix) with ESMTPS id C83B9C061356
        for <kvm@vger.kernel.org>; Thu, 11 Feb 2021 17:06:17 -0800 (PST)
Received: by mail-qv1-xf49.google.com with SMTP id q104so5230690qvq.20
        for <kvm@vger.kernel.org>; Thu, 11 Feb 2021 17:06:17 -0800 (PST)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=google.com; s=20161025;
        h=sender:reply-to:date:in-reply-to:message-id:mime-version:references
         :subject:from:to:cc;
        bh=S7QHW8ZiBK6tGJi1I7cAXOJQJ+RvuKnKUK7n46X1AcY=;
        b=wQ69e+BIoW8rAIQrjAViBirPdI9jogJOg0i0tgtI54ooUPlvlC6+fYYGwqY3JxmH4J
         es6Wm5RRcC5oMGlqG1M9ZDb/P7r+3Ft3EFL92/1Py9gkbmvuUOGOPicf3KC2Logks9qT
         yRKYsWngvUkpFKPNLLs5U8+qe4c3ik9gPWQKa9n1OI8nA72YeJq3/zj8hPKmjp2mvAC2
         qUMeMyk48r9Y0kl3FpDelUrotEitYv5WYh77HNeRs0BxMv9/L+dgCwCbolQz+xT8LFds
         ozrJsWK8mQfS5uc2Hh4dVreKjS2RM6jltd81xksmWQKzT+ZjbqU7+zqI5pwwqJPPSlVT
         jkfQ==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20161025;
        h=x-gm-message-state:sender:reply-to:date:in-reply-to:message-id
         :mime-version:references:subject:from:to:cc;
        bh=S7QHW8ZiBK6tGJi1I7cAXOJQJ+RvuKnKUK7n46X1AcY=;
        b=RFPVLNJ9UFlaoWjLCzjBp9QCY9HIMh5hXJMLwn16f+9D8G7mEnT2Gd8PbqxsFZsAWY
         BNG72u5F45HAXSyUEPQ4UnC5HEXbInRB7yNutKSpg3xttulr8SKEj2g6OkVYaYv6Cwo4
         PinhI84IDc59yaYwDuFEvHhwxEAL+sF0QJnAXtiIMSMYiOLFTPS9+CoAj4kLo4Yelubg
         6HWjnS6/fQvxl4I8BIAUTa0V4q6QitzOLHvDm4cOwg/VR+sZgvvEvOHZtf3J73zyqo2E
         8dCodEHutl91Q7JhLII3MCaInTzRQavmM2tAnKkrmLM2IYkowDLAGl9rGUmviLrnLyaU
         av6w==
X-Gm-Message-State: AOAM533tFbILLTksgamWps3Ekf58EXSdTvOXKbXzOcTjenRlNc9jvri0
        Mx3gMDak/VXyomTzR9bz1Uh80Vl4tAk=
X-Google-Smtp-Source: 
 ABdhPJwT5Po3+gUb70jgaXrEPNNYP5V4Z9WmVJlEAXSbopB1f6K4SMQ4YU2U45rxmArjJv9s+UDWT2kgpUI=
Sender: "seanjc via sendgmr" <seanjc@seanjc798194.pdx.corp.google.com>
X-Received: from seanjc798194.pdx.corp.google.com
 ([2620:15c:f:10:f588:a708:f347:3ebb])
 (user=seanjc job=sendgmr) by 2002:a05:6214:1188:: with SMTP id
 t8mr583077qvv.51.1613091977016; Thu, 11 Feb 2021 17:06:17 -0800 (PST)
Reply-To: Sean Christopherson <seanjc@google.com>
Date: Thu, 11 Feb 2021 17:06:05 -0800
In-Reply-To: <20210212010606.1118184-1-seanjc@google.com>
Message-Id: <20210212010606.1118184-4-seanjc@google.com>
Mime-Version: 1.0
References: <20210212010606.1118184-1-seanjc@google.com>
X-Mailer: git-send-email 2.30.0.478.g8a0d178c01-goog
Subject: [kvm-unit-tests PATCH 3/4] x86: Add a testcase for !PCID && INVPCID
From: Sean Christopherson <seanjc@google.com>
To: Paolo Bonzini <pbonzini@redhat.com>
Cc: kvm@vger.kernel.org, Sean Christopherson <seanjc@google.com>
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org

Add an "official" testcase for a guest with PCID disabled but INVPCID
enabled, which is supported by SVM and will soon be supported by VMX.

Signed-off-by: Sean Christopherson <seanjc@google.com>
---
 x86/pcid.c        | 9 +++++++--
 x86/unittests.cfg | 5 +++++
 2 files changed, 12 insertions(+), 2 deletions(-)

diff --git a/x86/pcid.c b/x86/pcid.c
index 64efd05..527a4a9 100644
--- a/x86/pcid.c
+++ b/x86/pcid.c
@@ -68,7 +68,7 @@ report:
     report(passed, "Test on PCID when disabled");
 }
 
-static void test_invpcid_enabled(void)
+static void test_invpcid_enabled(int pcid_enabled)
 {
     int passed = 0, i;
     ulong cr4 = read_cr4();
@@ -93,6 +93,10 @@ static void test_invpcid_enabled(void)
             goto report;
     }
 
+    /* Skip tests that require the PCIDE=1 if PCID isn't supported. */
+    if (!pcid_enabled)
+        goto success;
+
     if (write_cr4_checking(cr4 | X86_CR4_PCIDE) != 0)
         goto report;
 
@@ -103,6 +107,7 @@ static void test_invpcid_enabled(void)
     if (invpcid_checking(2, &desc) != 0)
         goto report;
 
+success:
     passed = 1;
 
 report:
@@ -139,7 +144,7 @@ int main(int ac, char **av)
         test_pcid_disabled();
 
     if (invpcid_enabled)
-        test_invpcid_enabled();
+        test_invpcid_enabled(pcid_enabled);
     else
         test_invpcid_disabled();
 
diff --git a/x86/unittests.cfg b/x86/unittests.cfg
index c3a4ee0..d8ef717 100644
--- a/x86/unittests.cfg
+++ b/x86/unittests.cfg
@@ -252,6 +252,11 @@ file = pcid.flat
 extra_params = -cpu qemu64,-pcid,-invpcid
 arch = x86_64
 
+[pcid-asymmetric]
+file = pcid.flat
+extra_params = -cpu qemu64,-pcid,+invpcid
+arch = x86_64
+
 [rdpru]
 file = rdpru.flat
 extra_params = -cpu max

From patchwork Fri Feb 12 01:06:06 2021
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Sean Christopherson <seanjc@google.com>
X-Patchwork-Id: 12084557
Return-Path: <kvm-owner@kernel.org>
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on
	aws-us-west-2-korg-lkml-1.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-26.3 required=3.0 tests=BAYES_00,DKIMWL_WL_MED,
	DKIM_SIGNED,DKIM_VALID,DKIM_VALID_AU,HEADER_FROM_DIFFERENT_DOMAINS,
	INCLUDES_CR_TRAILER,INCLUDES_PATCH,MAILING_LIST_MULTI,SPF_HELO_NONE,SPF_PASS,
	USER_AGENT_GIT,USER_IN_DEF_DKIM_WL autolearn=ham autolearn_force=no
	version=3.4.0
Received: from mail.kernel.org (mail.kernel.org [198.145.29.99])
	by smtp.lore.kernel.org (Postfix) with ESMTP id C3EBCC43332
	for <kvm@archiver.kernel.org>; Fri, 12 Feb 2021 01:07:13 +0000 (UTC)
Received: from vger.kernel.org (vger.kernel.org [23.128.96.18])
	by mail.kernel.org (Postfix) with ESMTP id 88BCC64E44
	for <kvm@archiver.kernel.org>; Fri, 12 Feb 2021 01:07:13 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S230159AbhBLBHA (ORCPT <rfc822;kvm@archiver.kernel.org>);
        Thu, 11 Feb 2021 20:07:00 -0500
Received: from lindbergh.monkeyblade.net ([23.128.96.19]:35572 "EHLO
        lindbergh.monkeyblade.net" rhost-flags-OK-OK-OK-OK) by vger.kernel.org
        with ESMTP id S230110AbhBLBGt (ORCPT <rfc822;kvm@vger.kernel.org>);
        Thu, 11 Feb 2021 20:06:49 -0500
Received: from mail-yb1-xb49.google.com (mail-yb1-xb49.google.com
 [IPv6:2607:f8b0:4864:20::b49])
        by lindbergh.monkeyblade.net (Postfix) with ESMTPS id 2841EC06121C
        for <kvm@vger.kernel.org>; Thu, 11 Feb 2021 17:06:20 -0800 (PST)
Received: by mail-yb1-xb49.google.com with SMTP id c12so7995276ybf.1
        for <kvm@vger.kernel.org>; Thu, 11 Feb 2021 17:06:20 -0800 (PST)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=google.com; s=20161025;
        h=sender:reply-to:date:in-reply-to:message-id:mime-version:references
         :subject:from:to:cc;
        bh=Y67LULj4Vit9d/qVGY5s43+v1n+wHLvy3C+5YsJH90A=;
        b=Fytt+DKAjecPvEF52pkEHtGEevMU2fMASC/WkmnrC9upf3i0sHXnDJZVK+a5L4+bLn
         nRR0ypVQy+3giJt/LDuKQQiK0uo/Y57HHC47ElOryxPXct9nb7NmcNnEfC1DmANFrFxV
         meSh14TwQ5FdEERq453r6pAPFvTMiy18Wr66leLoHbu6I/GE/sHTHKjJ0HP3MdMhfDkP
         zz0Buwa/hGWJSo4mNutskl0sKCXvKoHOMGmfR+s/5VxBtaxrxdVf5s/hXYJ4AM9n6c/y
         1QFsANpMOFiZsqjfAelZllccbGDBTdCzrlMnRftw2kiRa8JvdpCE56GdGNskprsyeden
         YhXg==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20161025;
        h=x-gm-message-state:sender:reply-to:date:in-reply-to:message-id
         :mime-version:references:subject:from:to:cc;
        bh=Y67LULj4Vit9d/qVGY5s43+v1n+wHLvy3C+5YsJH90A=;
        b=HtmtXsOM8iVMbaofGG9sqDCSfpnUw4zNSr7REhwcJ/ParS/CSY+otpSFGo+Paqd+nH
         lSPPd3urthqbthRBB16V1hSJOEL6y9l5GupFDB7LDzPDarNSBEJ66zGGOpW/LDKLBxFK
         husvzCFI0xt7SIniZvTD23CfI8/Yf8lJeXsDXs2axiJwNDQmvOAIcp2Fa0T8AI13TomT
         5+jgtolzoCOVqd+SInD4sf083jvIdJvctoKZG00UAcoFpEZY0c7sF83eGbkGj0a1zzhc
         D91IuoE54AB/PfntC67jBKiHIDUV5Xxx83sgwiv6saVHQOhQMD6EtHqldNx8LTScihu8
         7fsQ==
X-Gm-Message-State: AOAM533q5OgjD+vJ5wfb9rFxzhHtDTzGGEIuXoLo1gRbJ7Mdo0FLZRUM
        ZrGMh11IutrVOHx0FtXG+dGDTY5d2rw=
X-Google-Smtp-Source: 
 ABdhPJx7/o/ro7Frg69y6RXlpoGaTs+tQKO/6wykqxh8rniGmBgnmS1frYM1Wo8ojF784ZzxRZHUgFK4sfo=
Sender: "seanjc via sendgmr" <seanjc@seanjc798194.pdx.corp.google.com>
X-Received: from seanjc798194.pdx.corp.google.com
 ([2620:15c:f:10:f588:a708:f347:3ebb])
 (user=seanjc job=sendgmr) by 2002:a05:6902:1025:: with SMTP id
 x5mr832686ybt.21.1613091979443; Thu, 11 Feb 2021 17:06:19 -0800 (PST)
Reply-To: Sean Christopherson <seanjc@google.com>
Date: Thu, 11 Feb 2021 17:06:06 -0800
In-Reply-To: <20210212010606.1118184-1-seanjc@google.com>
Message-Id: <20210212010606.1118184-5-seanjc@google.com>
Mime-Version: 1.0
References: <20210212010606.1118184-1-seanjc@google.com>
X-Mailer: git-send-email 2.30.0.478.g8a0d178c01-goog
Subject: [kvm-unit-tests PATCH 4/4] x86: Add a 'pcid' group for the various
 PCID+INVPCID permutations
From: Sean Christopherson <seanjc@google.com>
To: Paolo Bonzini <pbonzini@redhat.com>
Cc: kvm@vger.kernel.org, Sean Christopherson <seanjc@google.com>
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org

Rename the 'pcid' test to 'pcid-enabled' and add a 'pcid' group to allow
easily running all three tests.

Signed-off-by: Sean Christopherson <seanjc@google.com>
---
 x86/unittests.cfg | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/x86/unittests.cfg b/x86/unittests.cfg
index d8ef717..0698d15 100644
--- a/x86/unittests.cfg
+++ b/x86/unittests.cfg
@@ -242,20 +242,23 @@ file = kvmclock_test.flat
 smp = 2
 extra_params = --append "10000000 `date +%s`"
 
-[pcid]
+[pcid-enabled]
 file = pcid.flat
 extra_params = -cpu qemu64,+pcid,+invpcid
 arch = x86_64
+groups = pcid
 
 [pcid-disabled]
 file = pcid.flat
 extra_params = -cpu qemu64,-pcid,-invpcid
 arch = x86_64
+groups = pcid
 
 [pcid-asymmetric]
 file = pcid.flat
 extra_params = -cpu qemu64,-pcid,+invpcid
 arch = x86_64
+groups = pcid
 
 [rdpru]
 file = rdpru.flat
